src/util/settings_persister.js

```diff
@@ -209,7 +216,11 @@ export const readInitialState = () => {
         value = null;
       }
     } else if (field.type === 'boolean') {
-      value = value === 'true';
+      if (!value && field.default !== null) {
```

Comment:
This is the key line.  If the default is `true` or `false`, the value will be set to that.